python/ray/serve/common.py

```diff
@@ -24,6 +28,51 @@ class EndpointInfo:
     route: str
 
 
+class ApplicationStatus(str, Enum):
+    DEPLOYING = "DEPLOYING"
+    RUNNING = "RUNNING"
+    DEPLOY_FAILED = "DEPLOY_FAILED"
+
+
+@dataclass
+class ApplicationStatusInfo:
+    status: ApplicationStatus
+    message: str = ""
+    deployment_timestamp: float = 0
+
+    def __eq__(self, other):
+        if not isinstance(other, ApplicationStatusInfo):
+            return False
+
+        return (
+            self.status == other.status
+            and self.message == other.message
+            and self.deployment_timestamp == other.deployment_timestamp
+        )
+
+    def __str__(self):
+        return (
+            f"Status: {self.status}\n"
+            f"Deployment Timestamp: {self.deployment_timestamp}\n"
+            f"Message: {self.message}"
+        )
```

Comment:
do we need this? dataclass already generate repr. If it's about presentation logic, can you put that in CLI instead of here?

Reply:
The dataclass representations from `common.py` aren't used in the CLI, since the CLI uses Pydantic schemas instead. These are essentially internal data structures at the moment.

I found these string representations pretty helpful while debugging. The dataclass' default representation was hard to read since it lacked any whitespace or formatting. Since this piece of code doesn't have a high maintenance requirement, I'd prefer to leave it in for future debugging. What are your thoughts?

Reply:
if it's for debugging, then it shouldn't be in `__str__` to cause confusion. can you rename the method to `debug_string()` instead? 
Additionally, I would appreciate if we can unify the style between the two data structures. Either using `json.dumps(..., indent=4)` or some other way to make it more structured.

Reply:
Sounds good, I've renamed the method and unified the style with `json.dumps()`.